opencog/atomspace/Handle.cc

```diff
@@ -34,6 +34,8 @@ using namespace opencog;
 const Handle Handle::UNDEFINED(ULONG_MAX);
 
 Handle::Handle(AtomPtr atom) : _uuid(atom->_uuid), _ptr(atom) {}
+Handle::Handle(void* atom_ptr) :  _uuid( (*((AtomPtr*)atom_ptr))->_uuid ),
+                                    _ptr(*((AtomPtr*)atom_ptr)) {}
```

Comment:
Yes, you can. You may not want to, but you can. It is ugly and hacky as hell but the only way to do what you wanted. The atom_ptr is the address of the shared_ptr, not the address of the Atom. It is dereferenced to be the shared_ptr that is an AtomPtr. 

The names are confusing, I admit. It was originally called atom_ptr_ptr but lots of places call `AtomPtr` variables `atom` in the code, like the line above, 36.